core/src/main/java/com/datastax/oss/driver/api/core/type/codec/TypeCodecs.java

```diff
@@ -163,9 +374,114 @@
+   * @see #localTimestampAt(ZoneId)
+   */
+  @NonNull
+  public static TypeCodec<LocalDateTime> localTimestampAt(@NonNull ZoneId timeZone) {
+    return new LocalTimestampCodec(timeZone);
+  }
+
+  /**
+   * Returns a codec mapping CQL lists to Java object arrays. Serialization and deserialization of
+   * elements in the array is delegated to the provided element codec.
+   *
+   * <p>This method is not suitable for Java primitive arrays. Use {@link #BOOLEAN_ARRAY}, {@link
+   * #BYTE_ARRAY}, {@link #SHORT_ARRAY}, {@link #INT_ARRAY}, {@link #LONG_ARRAY}, {@link
+   * #FLOAT_ARRAY} or {@link #DOUBLE_ARRAY} instead.
+   */
+  @NonNull
+  public static <T> TypeCodec<T[]> arrayOf(@NonNull TypeCodec<T> elementCodec) {
+    return new ObjectArrayCodec<>(elementCodec);
+  }
+
+  /**
+   * Returns a codec mapping Java Enums to CQL ints, according to their {@linkplain Enum#ordinal()
+   * ordinals}.
+   *
+   * @see #enumNamesOf(Class)
+   */
+  @NonNull
+  public static <EnumT extends Enum<EnumT>> TypeCodec<EnumT> enumOrdinalsOf(
+      @NonNull Class<EnumT> enumClass) {
+    return new EnumOrdinalCodec<>(enumClass);
+  }
+
+  /**
+   * Returns a codec mapping Java Enums to CQL varchars, according to their programmatic {@linkplain
+   * Enum#name() names}.
+   *
+   * @see #enumOrdinalsOf(Class)
+   */
+  @NonNull
+  public static <EnumT extends Enum<EnumT>> TypeCodec<EnumT> enumNamesOf(
+      @NonNull Class<EnumT> enumClass) {
+    return new EnumNameCodec<>(enumClass);
+  }
+
+  /** Returns a codec wrapping another codec's Java type into {@link Optional} instances. */
+  @NonNull
+  public static <T> TypeCodec<Optional<T>> optionalOf(@NonNull TypeCodec<T> innerCodec) {
+    return new OptionalCodec<>(innerCodec);
+  }
+
+  /**
+   * Returns a codec that maps the given Java type to JSON strings, using a default Jackson JSON
+   * mapper to perform serialization and deserialization of JSON objects.
+   *
+   * @see <a href="http://wiki.fasterxml.com/JacksonHome">Jackson JSON Library</a>
+   */
+  @NonNull
+  public static <T> TypeCodec<T> json(@NonNull GenericType<T> javaType) {
+    return new JsonCodec<>(javaType);
+  }
+
+  /**
+   * Returns a codec that maps the given Java class to JSON strings, using a default Jackson JSON
+   * mapper to perform serialization and deserialization of JSON objects.
+   *
+   * @see <a href="http://wiki.fasterxml.com/JacksonHome">Jackson JSON Library</a>
+   */
+  @NonNull
+  public static <T> TypeCodec<T> json(@NonNull Class<T> javaType) {
+    return new JsonCodec<>(javaType);
+  }
+
+  /**
+   * Returns a codec that maps the given Java type to JSON strings, using the provided Jackson
+   * {@link JsonMapper} to perform serialization and deserialization of JSON objects.
+   *
+   * @see <a href="http://wiki.fasterxml.com/JacksonHome">Jackson JSON Library</a>
+   */
+  @NonNull
+  public static <T> TypeCodec<T> json(
+      @NonNull GenericType<T> javaType, @NonNull JsonMapper jsonMapper) {
```

Comment:
The exposure of `JsonMapper` causes strange revapi exceptions. We might want to revisit this. Besides, Jackson (non-shaded) is included in the core driver because of the Insights client, which may be removed in the future.

Reply:
Yeah I'm not a big fan of piggybacking on that dependency, indeed we don't know how long it will be there. There's also a chance thats someone uses an incompatible version.

How about moving JSON to the examples?

Reply:
I find it a bit disappointing because moving this to the examples module would require users to copy-paste code. I would like to see something more straightforward. I'm tempted to re-introduce the "extras" module for everything related to Json.

Reply:
After giving this some more thought, I take back my previous statement. I think it is still better for users if everything is available from driver-core. If we ever remove Insights, we could switch Jackson to an optional dependency.